sourcegraph/sourcegraph
# App: Fix vscode extension setup deep link flow

Reported by @taylorsperry [here](https://sourcegraph.slack.com/archives/C04PE4H6XGT/p1684937916196769)

In this PR Sourcegraph App setup flow doesn't interrupt auth UI page URL and you should be able to pass connect vscode extension to the app flow successfully. 

https://github.com/sourcegraph/sourcegraph/assets/18492575/53c2a3f8-8520-4394-82e8-1e95cc4d8dad

## Follow up 
We discussed this problem with engs and @almeidapaulooliveira, and as soon as we have a new-new setup flow for the app, the logic will be
- When users click to connect, will redirect to the authorization page in the App.
- The authorisation page will return to VSCode, now connected, and display the chat.
- The wizard should redirect to the last page of the wizard.
![Screenshot 2023-05-24 at 1 22 04 PM](https://github.com/sourcegraph/sourcegraph/assets/18492575/8b2df747-2547-4693-9057-12dc013c4689)

## Test plan
- Run local app production build  `CODESIGNING=0 ./enterprise/dev/app/build.sh`
- Follow the setup flow 
- On the extension page, click vs code extension 
- Install the extension and try to connect it with Sourcegraph App
- You should see auth UI 
- After you click authorise, you should see a focused vs code 
- The app itself should show the home page (setup flow should be finished)
